addons/mail/static/tests/discuss_app/inbox.test.js

```diff
@@ -447,6 +454,41 @@ test("inbox: mark as read should not display jump to present", async () => {
     await contains("[title='Jump to Present']", { count: 0 });
 });
 
+test("inbox: can mark as read when received a message from the record without access to", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["res.partner"].create({ name: "General" });
+    const messageId = pyEnv["mail.message"].create({
+        body: "not empty",
+        model: "res.partner",
+        needaction: true,
+        res_id: channelId,
+    });
+    pyEnv["mail.notification"].create({
+        mail_message_id: messageId,
+        notification_type: "inbox",
+        res_partner_id: serverState.partnerId,
+    });
+    onRpc("/mail/inbox/messages", async (req) => {
+        const res = await discuss_inbox_messages.bind(MockServer.current)(req);
+        const new_res = {
+            data: {
+                "mail.notification": res.data["mail.notification"],
+            },
+            messages: res["messages"],
+        };
+        return new_res;
+    });
```

Comment:
I don't know if you could `onRpc()` without having to import the actual function of the route.
If this doesn't work, there's a `onRpcAfter` in the mail test helpers that allow to do it with minimal footprint.